app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionAndIssuesChartLegend.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import { redirect } from 'redux-first-router';
+import { injectIntl, intlShape } from 'react-intl';
+import { defectTypesSelector } from 'controllers/project';
+import { connect } from 'react-redux';
+import { activeProjectSelector } from 'controllers/user';
+import commonLegendStyles from '../common/legend/legend.scss';
+import chartStyles from './launchExecutionAndIssueStatistics.scss';
+import { getItemColor, getItemName, getItemNameConfig } from '../common/utils';
+
+const commonCx = classNames.bind(commonLegendStyles);
+const chartCx = classNames.bind(chartStyles);
+
+@injectIntl
+@connect(
+  (state) => ({
+    project: activeProjectSelector(state),
+    defectTypes: defectTypesSelector(state),
+  }),
+  {
+    redirect,
+  },
+)
+export class Legend extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    items: PropTypes.array,
+    defectTypes: PropTypes.object.isRequired,
+    project: PropTypes.string.isRequired,
+    onClick: PropTypes.func,
+    onMouseOver: PropTypes.func,
+    onMouseOut: PropTypes.func,
+  };
+
+  static defaultProps = {
+    items: [],
+    onClick: () => {},
+    onMouseOver: () => {},
+    onMouseOut: () => {},
+  };
+
+  state = {
+    uncheckedIds: [],
+  };
+
+  onClick = ({ currentTarget }) => {
```

Comment:
I don't see any usages of this function in this file.

Reply:
removed